homeassistant/components/guardian/button.py

```diff
@@ -0,0 +1,81 @@
+"""Buttons for the Elexa Guardian integration."""
+from __future__ import annotations
+
+from aioguardian import Client
+from aioguardian.errors import GuardianError
+
+from homeassistant.components.button import ButtonDeviceClass, ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import EntityCategory, EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import ValveControllerEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
+
+BUTTON_KIND_REBOOT = "reboot"
+BUTTON_KIND_RESET_VALVE_DIAGNOSTICS = "reset_valve_diagnostics"
+
+BUTTON_DESCRIPTIONS = (
+    EntityDescription(
+        key=BUTTON_KIND_REBOOT,
+        name="Reboot",
+    ),
+    EntityDescription(
+        key=BUTTON_KIND_RESET_VALVE_DIAGNOSTICS,
+        name="Reset valve diagnostics",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Guardian buttons based on a config entry."""
+    async_add_entities(
+        [
+            GuardianButton(
+                entry,
+                hass.data[DOMAIN][entry.entry_id][DATA_CLIENT],
+                hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR],
```

Comment:
Unrelated to this pr, but it would be nice if `hass.data[DOMAIN][entry.entry_id]` was a dataclass so you could do

`data: TYPE = hass.data[DOMAIN][entry.entry_id]`

Then you could do

`data.client`
and 
`data.coordinator`

here